src/components/authenticationComponents/SignUpForm.js

```diff
@@ -1,67 +1,72 @@
-import { React, useState } from "react";
-import { Form, Input, Button, InputNumber, Select, message } from "antd";
+import { React } from "react";
+import { Form, Input, Button, Select, message } from "antd";
 import css from "./design/SignUpForm.module.css";
-import { LoadingOutlined } from "@ant-design/icons";
-import { useGlobalContext } from "../../context";
-import { useLocation } from "react-router-dom";
-import axios from "axios";
-const singUpAPI = "http://localhost:8001/signup";
+import postRequest from "../../API/postRequest";
+import { getAllCountries, getAllStates, getAllCities } from "../../actions/fetchCountryActions";
+import { useState } from "react";
+import { useEffect } from "react";
+import { useSelector, useDispatch } from "react-redux";
 
-export default function SignUpForm({ setIsModalOpen, setActiveTab }) {
-   const [form] = Form.useForm();
-   const { states, country, city, fetchState, fetchCity } = useGlobalContext();
+export default function SignUpForm({ setActiveTab }) {
+   const phoneNumberRegex = /^\(?\d{3}\)?[-.\s]?\d{3}[-.\s]?\d{4}$/;
+   const passwordRegex = /^(?=.*[A-Z])(?=.*[!@#$%^&*])(?=.*[0-9]).{8,}$/;
+   const zipcodeRegex = /\d{6}/;
```

Comment:
make utils folder and add all utilities there and use it here by exporting it from there